iterator.js

```diff
@@ -54,7 +46,6 @@ Iterator.prototype._next = function (callback) {
 }
 
 Iterator.prototype._end = function (callback) {
-  delete this.cache
```

Comment:
I see tests failed on checking the cache size after calling `.end()`, is there a particular reason to manually delete the cache array here?

AFAIK, deleting properties of a class is not recommended practice sine it might affect V8's hidden class.